app/assets/src/components/views/SampleUploadFlow/ReviewStep.jsx

```diff
@@ -43,10 +66,27 @@ class ReviewStep extends React.Component {
       errorMessage: "",
     });
 
-    // For uploading samples with files on S3
-    if (this.props.uploadType === "remote") {
-      bulkUploadRemote({
-        samples: this.props.samples,
+    // For uploading samples with files on S3 or Basespace
+    if (
+      this.props.uploadType === "remote" ||
+      this.props.uploadType === "basespace"
+    ) {
+      let bulkUploadFn = bulkUploadRemote;
+      let bulkUploadFnName = "bulkUploadRemote";
+      let samples = this.props.samples;
+      let successLogEvent = "ReviewStep_remote-upload_succeeded";
+      let failureLogEvent = "ReviewStep_remote-upload_failed";
+
+      if (this.props.uploadType === "basespace") {
+        bulkUploadFn = bulkUploadBasespace;
+        bulkUploadFnName = "bulkUploadBasespace";
+        samples = map(pick(BASESPACE_SAMPLE_FIELDS), samples);
+        successLogEvent = "ReviewStep_basespace-upload_succeeded";
```

Comment:
Not sure why attributes are better than events. It seems like events are slightly easier (since you only need to query on the event name, instead of the event AND the attribute). Happy to change if needed.

Reply:
Depends how you aggregate them or think about analysis. Normally, I would be interested in seeing successful vs. failed uploads, first, and then slice by type of upload. In the end, I see the core event is sample uploaded. 
As I said, there are only two variations here (although there is the local upload one), and it could become more relevant if we added more upload types...
Think of a case where you might have more types... that could lead to an explosion of event with weirder queries (E.g. either `event IN ['ReviewStep_type1', 'ReviewStep_type2', ... 'ReviewStep_type3']` or `event LIKE 'ReviewStep_%'`... 
As with any decision is a trade-off... and should have the analysis you are planning into account.